### PR TITLE
Model tests

### PR DESCRIPTION
**Pull request by elioncho**

### CODE REVIEW

File: test/models/game_test.rb [hunk 1]
```diff
@@ -1,7 +1,121 @@
 require 'test_helper'
 
 class GameTest < ActiveSupport::TestCase
-  # test "the truth" do
-  #   assert true
-  # end
+  describe 'initialize' do
```
Comment (EdgarOrtegaRamirez) [inline test/models/game_test.rb]:
```ruby
describe Game do
  describe "initialize" do
  end
  describe "validations" do
  end
  describe "callbacks" do
  end
end
```

File: Gemfile [hunk 2]
```diff
@@ -57,6 +60,7 @@ gem 'devise-bootstrap-views'
 gem 'devise-i18n'
 gem 'rubocop', require: false
 gem 'kaminari'
+gem "minitest-rails"
 gem 'rails-erd'
+gem 'simplecov', :require => false, :group => :test
```
Comment (EdgarOrtegaRamirez) [inline Gemfile]: This belongs to the `:test` group.

File: Gemfile [hunk 3]
```diff
@@ -57,6 +60,7 @@ gem 'devise-bootstrap-views'
 gem 'devise-i18n'
 gem 'rubocop', require: false
 gem 'kaminari'
+gem "minitest-rails"
```
Comment (EdgarOrtegaRamirez) [inline Gemfile]: This belongs to the `:development and :test` group.

File: app/models/move.rb [hunk 4]
```diff
@@ -3,23 +3,28 @@ class Move < ApplicationRecord
...
   belongs_to :game_ship, optional: true
 
-  before_validation :set_row_and_column
+  before_validation :set_row_and_column, if: :coordinates?
   after_create :update_game_state
 
   private
 
+  def coordinates?
+    !coordinates.blank?
```
Comment (EdgarOrtegaRamirez) [inline app/models/move.rb]: `coordinates.present?`

File: app/models/move.rb [hunk 5]
```diff
@@ -38,6 +43,6 @@ def set_row_and_column
   end
 
   def game_is_on_progress
-    errors.add(:base, 'El juego ya ha terminado.') if game.finished?
+    errors.add(:base, 'El juego ya ha terminado.') if game && game.finished?
```
Comment (EdgarOrtegaRamirez) [inline app/models/move.rb]: a move should always belong to a game, please remove this check.

File: test/fixtures/ships.yml [hunk 6]
```diff
@@ -1,9 +1,20 @@
 # Read about fixtures at http://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
```
Comment (EdgarOrtegaRamirez) [inline test/fixtures/ships.yml]: Remove fixtures

File: Gemfile (hunk 2 above)
Comment (elioncho) [inline Gemfile]: Ahí ya está en el test group, lo meto en el development y test group?

File: test/factories/games.rb [hunk 7]
```diff
@@ -1,5 +1,5 @@
 FactoryGirl.define do
   factory :game do
-
+    association :user, factory: :user
```
Comment (EdgarOrtegaRamirez) [inline test/factories/games.rb]: Simplify this, see http://www.rubydoc.info/gems/factory_girl/file/GETTING_STARTED.md#Associations for more information.

**Pull request merged**